### PR TITLE
172 finishline

### PR DESCRIPTION
**Pull request by snuggs**

### CODE REVIEW

File: com.finishline/cart.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        expect( cart.products ).to.exist
+        expect( cart.totals.totalQuantity ).to.exist
+        expect( cart.totals.subtotal ).to.match(/^\$[0-9]+\.[0-9]{2}$/)
+
+        done()
+      })
+    })
+  })
+
+  describe.only('.add(item)', function () {
```
Comment (snuggs) [inline com.finishline/cart.spec.js]: UGHH!!!

File: com.finishline/product.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+module.exports.prototype.map_sizes_to_skus = function () {
+  var self = this
+
+  return Promise.all(_skus.map(function (sku) {
+    return sku_to_size.call(self, sku)
+  }))
+}
+
+module.exports.prototype.select = function (size) {
```
Comment (snuggs) [inline com.finishline/product.js]: this method can be refactored.

File: com.finishline/cart.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  return deferred.promise
+}
+
+module.exports.prototype.add = function (item) {
+  var deferred = Promise.pending()
+    , url = 'http://www.finishline.com/store/global/fragments/addItemToDynamicCart.jsp'
+//  , url = 'http://www.finishline.com/store/product?productId=prod783903'
+
+  this._agent.get(url)
+  .set('Referer', 'http://www.finishline.com/store/product?productId=prod' + '783903')
```
Comment (snuggs) [inline com.finishline/cart.js]: Should not be hard set. Must be the productId of ````item````

**Comment by RobertChristopher** [comment]
Looks real good to me. @snugs Nice work.

**Pull request merged**